internal/expr/typeinfo.go

```diff
@@ -13,12 +13,18 @@ type typeMember interface {
 	outerType() reflect.Type
 }
 
+type sliceInfo struct {
+	length int
+}
+
 type mapKey struct {
 	name    string
 	mapType reflect.Type
+	// sliceInfo is nil if the map value is not allowed to be a slice.
+	slice *sliceInfo
```

Comment:
The `nil` semantics not being reflected in the field name feels messy. Perhaps renaming this field to something like `allowedSliceInfo` would help?

Reply:
Fair, changed